### PR TITLE
Add a lint for SuccessOrExit without assignment.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
This is almost always a mistake, since it loses track of the error involved.

### CODE REVIEW

File: src/access/AccessControl.cpp [hunk 1]
```diff
@@ -608,6 +609,10 @@ bool AccessControl::IsValid(const Entry & entry)
     return true;
 
 exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DataManagement, "AccessControl: %s %" CHIP_ERROR_FORMAT, log, err.Format());
+    }
     ChipLogError(DataManagement, "AccessControl: %s", log);
```
Comment (Damian-Nordic) [inline src/access/AccessControl.cpp:616]: This log is not redundant. The previous code was correct, but it didn't log details of an error.

**Review by Damian-Nordic — APPROVED** [review]

File: src/access/AccessControl.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply src/access/AccessControl.cpp:616]: Er, right, I assume you mean "now redundant".

We can still end up here when `err == CHIP_NO_ERROR` and need to log things.  I will fix that up.

Comment (bzbarsky-apple) [inline src/access/AccessControl.cpp:616]:
```suggestion
    else
    {
        ChipLogError(DataManagement, "AccessControl: %s", log);
    }
```

**Review by yufengwangca — APPROVED** [review]

File: src/access/AccessControl.cpp (hunk 1 above)
Comment (Damian-Nordic) [reply src/access/AccessControl.cpp]: Yes, "now redundant" :)

**Comment by dhrishi** [comment]
@bzbarsky-apple Is it possible to cherry-pick this fix in v1.1-branch?

**Comment by bzbarsky-apple** [comment]
@dhrishi Probably?  I don't have time/bandwidth for it right now, but if you try and it fails let me know and I can try to make the time.

**Comment by dhrishi** [comment]
@bzbarsky-apple Sure. Will do that!

**Pull request merged**